WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -134,23 +134,25 @@ private void changeLanguage(String languageCode) {
 
         Resources res = getResources();
         Configuration conf = res.getConfiguration();
-        Locale currentLocale = conf.locale != null ? conf.locale : LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext());
+        Locale currentLocale = conf.locale != null ?
+                conf.locale : LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext());
 
-        if (currentLocale.toString().equals(languageCode)) return;
-
-        updateLanguagePreference(languageCode);
+        if (currentLocale.getLanguage().equals(new Locale(languageCode).getLanguage())) {
+            return;
+        }
 
-        // update configuration
         Locale newLocale = WPPrefUtils.languageLocale(languageCode);
-        conf.locale = newLocale;
-        res.updateConfiguration(conf, res.getDisplayMetrics());
-
-        if (LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext()).equals(newLocale)) {
+        if (currentLocale.getLanguage().equals(newLocale.getLanguage())) {
```

Comment:
This comparison doesn't work because `currentLocale` isn't the current device language unless `conf` is null above (which I think it would rarely be).